modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Utils.java

```diff
@@ -130,8 +132,45 @@ public static BytesReference toBytesReference(final ByteBuf buffer) {
         }
     }
 
+    /**
+     * Wrap Netty's {@link ByteBuf} into {@link ReleasableBytesReference} and delegating reference count to ByteBuf.
+     */
     public static ReleasableBytesReference toReleasableBytesReference(final ByteBuf buffer) {
-        return new ReleasableBytesReference(toBytesReference(buffer), buffer::release);
+        return new ReleasableBytesReference(toBytesReference(buffer), toRefCounted(buffer));
+    }
+
+    public static RefCounted toRefCounted(final ReferenceCounted ref) {
```

Comment:
We already have this in `Netty4MessageInboundHandler.ByteBufRefCounted` lets reuse that :)

Reply:
Sure, does it make sense to bring ByteBufRefCounted into Netty4Utils?

Reply:
Thanks! Addressed in current revision.

Reply:
Also updated PR description that it's something we already do in Transport, not a new thing.